addons/l10n_fr_pos_cert/static/src/js/l10n_fr_pos_cert.js

```diff
@@ -0,0 +1,244 @@
+/******************************************************************************
+    Copyright (C) 2017 - Today: GRAP (http://www.grap.coop)
+    @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+    Part of Odoo. See LICENSE file for full copyright and licensing details.
+ *****************************************************************************/
+
+/******************************************************************************
+    Developpers Note: 
+    This module changes the behaviour of the PoS regarding the French Sapin Law.
+    In this context, for french Point of Sales, when a pos order is created
+    a call to the server will be done, to recover the hash of the order to
+    print it on the bill.
+    The time for the bill to print will depend on the server processing time.
+
+    1. If you want to change the message printed on the bill, overload the
+    function prepare_certification_text()
+
+    2. If you want to change the behaviour of the PoS if server is unreachable,
+    (for exemple, prevent printing the bill), overload the following functions :
+    - ReceiptScreenWidget.show_certification()
+    - ProxyDevice.print_receipt_certification()
+******************************************************************************/
+
+'use strict';
+
+odoo.define('l10n_fr_pos_cert.models', function (require) {
+
+    var devices = require('point_of_sale.devices');
+    var models = require('point_of_sale.models');
+    var screens = require('point_of_sale.screens');
+    var DataModel = require('web.DataModel');
+    var core = require('web.core');
+    var _t = core._t;
+
+    /*************************************************************************
+        Promise that will be resolved, when the hash of the saved order is
+        known
+     */
+    var certification_deferred = null;
```

Comment:
I would prefer certification_deferred to be a member of this.pos in Models, Devices and PosBaseWidgets